### PR TITLE
ci: refine the github actions

### PR DESCRIPTION
**Pull request by zyy17**
## Summary by CodeRabbit

- **New Features**
  - Enhanced flexibility in PR title formatting, allowing an exclamation mark to be used independently of scope.
  - Updated GitHub Actions workflow names for improved clarity.

- **Improvements**
  - Upgraded Go version to 1.22 across multiple workflows for access to the latest features and fixes.
  - Updated actions in workflows to their latest versions, ensuring improved functionality.

- **Bug Fixes**
  - Streamlined commit handling in the development build workflow for better usability.

### CODE REVIEW

**Comment by zyy17** [comment]
@daviderli614 I change the image tag from `*.*.*` to `v*.*.*` to keep consistent with greptimedb convention.

**Review by daviderli614 — APPROVED** [review]
LGTM

**Comment by daviderli614** [comment]
This script needs to be updated：https://github.com/GreptimeTeam/helm-charts/blob/main/scripts/update-chart.sh#L33

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -63,7 +63,6 @@ jobs:
         shell: bash
         run: |
           VERSION=${{ github.ref_name }}
```
Comment (daviderli614) [inline .github/workflows/release.yaml:65]: Need to write VERSION to environment variable?

Comment (daviderli614) [reply .github/workflows/release.yaml:65]: https://github.com/GreptimeTeam/greptimedb-operator/blob/develop/.github/workflows/release.yaml#L78

Comment (zyy17) [reply .github/workflows/release.yaml:65]: Let's keep it to make it easy to configure in the future.

**Pull request merged**